The generation of the amd64 dashboard package is failing during the testing stage

## Description

While generating v4.10.3-rc1 packages in https://github.com/wazuh/wazuh-qa-automation/issues/3153, an error occurred while generating the amd64 dashboard package during the `Test package` stage.

The error indicates that the path `wazuh-dashboard/dev-tools/build-packages/output/deb` does not exist:

```
ls: cannot access '/home/runner/work/wazuh-dashboard/wazuh-dashboard/dev-tools/build-packages/output/deb': No such file or directory
Error: Process completed with exit code 2.
```

Failed action: https://github.com/wazuh/wazuh-dashboard/actions/runs/16834011526/job/47790492344

--- a/dev-tools/build-packages/deb/docker/amd64/Dockerfile
+++ b/dev-tools/build-packages/deb/docker/amd64/Dockerfile
@@ -1,4 +1,4 @@
-FROM debian:10
+FROM debian:12
 
 ENV DEBIAN_FRONTEND noninteractive
 

